modules/analysis/src/service.rs

```diff
@@ -0,0 +1,1024 @@
+                            }
+                            let p2 = match nodes.get(&right_purl_string) {
+                                Some(&i) => i,
+                                None => {
+                                    // Create a Purl from the left_purl_string
+                                    let right_purl = match Purl::from_str(&right_purl_string) {
+                                        Ok(purl) => purl,
+                                        Err(e) => {
+                                            log::warn!("Error parsing Purl: {}", e);
+                                            return; // TODO: maybe we do more here
+                                        }
+                                    };
+                                    let new_node = PackageNode {
+                                        purl: right_purl_string.clone(),
+                                        name: right_purl.name.clone(),
+                                        published: sbom_published.clone(),
+                                        document_id: document_id.clone(),
+                                        product_name: product_name.clone(),
+                                        product_version: product_version.clone(),
+                                    };
+                                    let i = g.add_node(new_node);
+                                    nodes.insert(right_purl_string.clone(), i);
+                                    i
+                                }
+                            };
+
+                            g.add_edge(p1, p2, relationship);
+                        }
+
+                        graph_write_gaurd.insert(distinct_sbom_id.to_string(), g);
+                    }
+                    Err(err) => {
+                        log::error!("Error fetching graph relationships: {}", err);
+                    }
+                }
+            }
+        }
+    }
+}
+
+impl AnalysisService {
+    pub fn new(db: Database) -> Self {
+        GraphMap::get_instance();
+        Self { db }
+    }
+
+    pub async fn load_graphs<TX: AsRef<Transactional>>(
+        &self,
+        distinct_sbom_ids: Vec<String>,
+        tx: TX,
+    ) -> Result<(), Error> {
+        let connection = self.db.connection(&tx);
+        load_graphs(&connection, &distinct_sbom_ids).await;
+
+        Ok(())
+    }
+    pub async fn load_all_graphs<TX: AsRef<Transactional>>(&self, tx: TX) -> Result<(), Error> {
+        let connection = self.db.connection(&tx);
+        // retrieve all sboms in trustify
+        let distinct_sbom_ids: Vec<String> = sbom::Entity::find()
+            .select()
+            .order_by(sbom::Column::DocumentId, Order::Asc)
+            .order_by(sbom::Column::Published, Order::Desc)
+            .all(&connection)
+            .await?
+            .into_iter()
+            .map(|record| record.sbom_id.to_string()) // Assuming sbom_id is of type String
```

Comment:
comment: was unclear if we should use sbom_id::uuid or something like document_id::string